server.go

```diff
@@ -22,6 +24,8 @@ const (
 var (
 	ErrorServerNotSet       = errors.New("Server not set")
 	ErrorConnectionNotFound = errors.New("Connection not found")
+
+	ErrRateLimiting = errors.New("gosocketio: Rate limiting reached, a message was dropped")
```

Comment:
I don't know if I love the idea of just dropping/ignoring a message because of the rate limiter.. but I guess that's the only way to prevent a lot of additional resources being attributed. Unless the goroutine was started, but didn't make any progress outside of stating a goroutine and had a timeout / order of execution rules.

Reply:
Correct, the error is to indicate a goroutine did not start because a rate limit was hit. Since it wasn't started there isn't anything to do with the message.

Instead when the rate limit is hit it calls the ErrorHandler with that error, so at least we know it happened. But we don't know on what message. Without rewriting more it was sorta difficult to do another way. @IronSinew did you have another suggestion?